### ISSUE TITLE
Multiple lambdas using same image creates duplicate ECR+image

### ISSUE THREAD

**Issue by grgghrn** [aws/aws-sam-cli#3750]
I found closed issue #2576 which I believe at that point this just wasn't working at all, but it is indeed working, and this is more of a nitpicky efficiency detail I wanted to see if was possible. 

I have 2 lambdas referencing the same Metadata image definitions (just different 'cmd's).  It works, but it ends up creating multiple ECR's (using --resolve-image-repos), duplicating the image even though it's the same.  So it's just an efficiency issue since (my upload speeds are pitiful from home!).  I know it works if you manually deploy the image and reference the same ImageUri in config (if not using Metadata) so it is technically feasible.

thanks

**Comment by qingchm** [comment]
Thanks for mentioning this improvement idea to us, we really appreciate your feedback and will get back to you if we feel that we can address this request!

**Comment by medley56** [comment]
I'd like to jump on here and request that this be addressed. One of the main benefits to dockerizing my lambdas is so that I can package all my lambda code into a single image and run multiple lambda handler commands all from the same image. There's no benefit if every lambda definition creates its own ECR and spends time rebuilding the same docker image over and over.

**Comment by jtanios** [comment]
Agreed. If you have 10 lambda functions all using an identical image, 10 duplicate images will be uploaded to ECR. Can SAM build & upload one image, and get all the lambdas to have the same `ImageUri`?

**Comment by chandlersong** [comment]
and updated for the issue？

**Comment by chandlersong** [comment]
this feature is very useful. Hope it can be added as soon as possible

**Comment by mayankkapoor** [comment]
Same with me. I publish two different functions with the same image (one is a producer, another a consumer). I need to push twice. Would be great if I can use a single push via SAMCLI to publish two different functions.

**Comment by ahmad-asmar** [comment]
Is this going to be fixed anytime soon? We are migrating from serverless framework and it's possible to define an image and use it in all functions but in SAM it duplicates the ECR creation and also push the image named after the first function to the other ECRs so if we have helloWorld and helloAgain functions the helloAgain ECR would have helloworld tagged image. This is a huge deal for us to use SAM. @qingchm is there any workaround for now to have proper image tag and not duplicating the ECR and the image push?

**Comment by iamtripathi25** [comment]
Hey!

Is anyone working to resolve this, Its literally hindering my provisioning time. SAM was actually meant to reduce provisioning time.
